apps/server/src/modules/room/api/test/room-get-room-invitation-link-list.api.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { schoolEntityFactory } from '@modules/school/testing';
+import { serverConfig, ServerConfig, ServerTestModule } from '@modules/server';
+import { HttpStatus, INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { cleanupCollections } from '@testing/cleanup-collections';
+import { UserAndAccountTestFactory } from '@testing/factory/user-and-account.test.factory';
+import { TestApiClient } from '@testing/test-api-client';
+import { roomEntityFactory } from '../../testing';
+
+describe('Room Invitation Link Controller (API)', () => {
+	let app: INestApplication;
+	let em: EntityManager;
+	let testApiClient: TestApiClient;
+	let config: ServerConfig;
+
+	beforeAll(async () => {
+		const moduleFixture = await Test.createTestingModule({
+			imports: [ServerTestModule],
+		}).compile();
+
+		app = moduleFixture.createNestApplication();
+		await app.init();
+		em = app.get(EntityManager);
+		testApiClient = new TestApiClient(app, 'rooms');
+
+		config = serverConfig();
+	});
+
+	beforeEach(async () => {
+		await cleanupCollections(em);
+		config.FEATURE_ROOM_INVITATION_LINKS_ENABLED = true;
+	});
+
+	afterAll(async () => {
+		await app.close();
+	});
+
+	describe('GET /:room-id/room-invitation-links', () => {
+		describe('when the user is not authenticated', () => {
+			it('should return a 401 error', async () => {
+				const someId = new ObjectId().toHexString();
+				const response = await testApiClient.get(someId);
+				expect(response.status).toBe(HttpStatus.UNAUTHORIZED);
+			});
+		});
+
+		describe('when the feature is disabled', () => {
+			const setup = async () => {
+				config.FEATURE_ROOM_INVITATION_LINKS_ENABLED = false;
+
+				const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent();
+				await em.persistAndFlush([studentAccount, studentUser]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(studentAccount);
+
+				return { loggedInClient };
+			};
+
+			it('should return a 403 error', async () => {
+				const { loggedInClient } = await setup();
+				const someId = new ObjectId().toHexString();
+				const response = await loggedInClient.get(`${someId}/room-invitation-links`);
+				expect(response.status).toBe(HttpStatus.FORBIDDEN);
+			});
+		});
+
+		describe('when id is not a valid mongo id', () => {
+			const setup = async () => {
+				const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent();
+				await em.persistAndFlush([studentAccount, studentUser]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(studentAccount);
+
+				return { loggedInClient };
+			};
+
+			it('should return a 400 error', async () => {
+				const { loggedInClient } = await setup();
+				const response = await loggedInClient.get('42/boards');
+				expect(response.status).toBe(HttpStatus.BAD_REQUEST);
+			});
+		});
+
+		describe('when the user has the required permissions', () => {
```

Comment:
within this describe, so we also want the case `when the room exists but has no links` to be tested?

Reply:
the indentation in the code and in the test summary makes it easy to understand. I would not use "and" here, but I am open for a discussion.

Reply:
done. moved it outside